src/deserializeSchema.ts

```diff
@@ -466,14 +528,36 @@ export function deserialUint32(source: Readable): number {
 }
 
 /**
- *
  * @param buffer wasm file buffer
+ * @param moduleVersion the version of the module, defaults to 0
  * @returns deserialized module of wasm file
  */
-export function deserialModuleFromBuffer(buffer: Buffer): Module {
+export function deserialModuleFromBuffer(
+    buffer: Buffer,
+    moduleVersion: 0
+): Module;
+export function deserialModuleFromBuffer(
+    buffer: Buffer,
+    moduleVersion: 1
+): ModuleV1;
+export function deserialModuleFromBuffer(
+    buffer: Buffer,
+    moduleVersion: number
+): Module | ModuleV1;
+export function deserialModuleFromBuffer(
```

Comment:
I understand that you want to leave `0` as the default to avoid a major version change, but concordium-client and cargo-concordium use smart contract version 1 by default, so this might be confusing for the users.
Consider making this a breaking change.

Reply:
Yeah, we'll do this as a major.
Though I have made the decision to not use a default. Then when we add support for higher versions, then we can have that as the default.
Also I assume this is not really used directly, but instead through the `serializeInitContractParameters` and `serializeUpdateContractParameters` functions.